src/components/map/MapService.js

```diff
@@ -881,6 +885,64 @@
               attribution: layer.attribution,
               layers: subLayers
             });
+          } else if (layer.type == 'geojson') {
+            var olSource = new ol.source.GeoJSON();
+            olLayer = new ol.layer.Vector({
+              source: olSource
+            });
+            var key, timer;
+            var geojsonFormat = new ol.format.GeoJSON();
+            var setLayerSource = function() {
+              var fullUrl = gaGlobalOptions.ogcproxyUrl + layer.geojsonUrl;
+              $http.get(fullUrl, {
+                cache: false
+              }).success(function(data) {
+                olSource.clear();
+                olSource.addFeatures(
+                  geojsonFormat.readFeatures(data)
+                );
+                // If map is not provided we don't want to add the timers
+                if (map) {
```

Comment:
I'm a little sceptic to have a map parameter for this function now. We should really try to avoid it (an ollayer should be independant of map). I think you mentioned the same concerns in your comments.

I would try to put this code either when we add/remove the layer, or in a separate service that 'watches' the map for changes and acts accordingly.